File: LC/[0217] Contains Duplicate.py
# one liner
class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        return True if len(set(nums)) < len(nums) else False

# sol 2
class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        visited=set()
        for n in nums:
            if n in visited: return True
            visited.add(n)
        return False

File: LC/[0217] Contains Duplicate.cpp
class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        unordered_set<int> visited;
        for(auto n: nums){
            if(visited.find(n) != visited.end()){
                return true;
            }
            visited.insert(n);
        }
        return false;
    }
};
